src/main/java/com/suse/saltstack/netapi/client/SaltStackClient.java

```diff
@@ -290,6 +290,37 @@ public Job call() throws SaltStackException {
     }
 
     /**
+     * Query for result of supplied job.
+     * 
+     * GET /job/<job-id>
+     * 
+     * @param job {@link Job} object representing scheduled job
+     * @return Map key: minion id, value: command result from that minion
+     * @throws SaltStackException 
+     */
+    public Map<String,Object> queryJobResult( final Job job ) throws SaltStackException{
+    	return queryJobResult(job.getJid());
+    }
+
+    /**
+     * Query for result of supplied job.
+     * 
+     * GET /job/<job-id>
+     * 
+     * @param job String representing scheduled job
+     * @return Map key: minion id, value: command result from that minion
+     * @throws SaltStackException 
+     */
+    public Map<String,Object> queryJobResult( final String job) throws SaltStackException{
+        Result<List<Map<String,Object>>> result = connectionFactory
+                .create("/jobs/"+job, JsonParser.RETVALS, config)
+                .getResult();
+
+        // A list with one element is returned, we take the first
+        return result.getResult().get(0);
```

Comment:
What will happen if result gets empty? Maybe:
```java
    ...
    return (!result.getResult().isEmpty() ? result.getResult().get(0) : null)
```

...or new empty List (depends). But certainly I would be worried about get something like "index out of range" error.